src/main/java/com/kodilla/ecommercee/service/UserDbService.java

```diff
@@ -0,0 +1,49 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.User;
+import com.kodilla.ecommercee.domain.UserNotFoundException;
+import com.kodilla.ecommercee.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class UserDbService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    public List<User> getAllUsers() {
+        return userRepository.findAll();
+    }
+
+    public Optional<User> getUser(final Long id) throws UserNotFoundException {
+        return userRepository.findById(id);
+    }
+
+    public void deleteUser(final long id) {
+        if (userRepository.existsById(id)) {
+            userRepository.deleteById(id);
+        }
+    }
+
+    public User saveUser(final User user) {
+        return userRepository.save(user);
+    }
+
+    public User createUser(final User user) {
+        return userRepository.save(user);
+    }
+
+    public void blockUser(final User user) {
+        if (userRepository.existsById(user.getId())) {
```

Comment:
Generalnie może tak zostać, ale jakbyś miał czas i możliwości to obejrzyj sobie:
https://www.youtube.com/watch?v=tKfDdtjYYi8
i zastanów się jakby to można było "inaczej" zapisać.
Na kursie nie poruszamy zagadnień związanych z transakcjami, ale warto mieć świadomość, że coś takiego jest i to coś w jakiś sposób wpływa na nasz kod